### PR TITLE
Omit headers and data for empty messages

### LINKED ISSUE #132: Web Push: Empty messages don't need crypto headers
IIRC, it's okay to omit the `Content-Encoding` and `Encryption` headers for push messages without a body. The DOM API can distinguish these from empty encrypted messages: `pushEvent.data === null` for the former; `pushEvent.data.text() === ""` for the latter.

### PR DESCRIPTION
**Pull request by kitcambridge**
This lets folks use Web Push as if it were Simple Push. For that case, I don't think there's much point to sending an encrypted empty string.

Closes #132.

### CODE REVIEW

File: autopush/tests/test_integration.py [hunk 1]
```diff
@@ -183,6 +188,9 @@ def get_notification(self, timeout=0.2):
         except:
             return None
 
+    def get_notification(self, timeout=0.2):
```
Comment (bbangert) [inline autopush/tests/test_integration.py]: What's the rationale for refactoring this into two methods?

**Comment by bbangert** [comment]
No changelog entry re: this change?

File: autopush/tests/test_integration.py (hunk 1 above)
Comment (kitcambridge) [inline autopush/tests/test_integration.py]: I wanted `send_notification` to accept a timeout, but I couldn't call `self.get_notification` directly—that caused  `__getattribute__` to wrap it. I also made a change on line 74 to ignore all `_`-prefixed methods. Is there a better way to bypass `__getattribute__`?

Comment (bbangert) [inline autopush/tests/test_integration.py]: Yup, the same way __getattribute__ bypasses it, grab the actual one with:
 ``get = object.__getattribute__(self, "get_notification")```

Comment (kitcambridge) [inline autopush/tests/test_integration.py]: D'oh, I forgot `__getattribute__` is inherited and you can actually do that. :stuck_out_tongue_winking_eye:

**Comment by bbangert** [comment]
r+

**Pull request merged**